src/components/ble/SimpleWeatherService.cpp

```diff
@@ -153,10 +168,28 @@ std::optional<SimpleWeatherService::Forecast> SimpleWeatherService::GetForecast(
   return {};
 }
 
+bool SimpleWeatherService::IsNight() const {
+  if (currentWeather && currentWeather->sunrise > 0 && currentWeather->sunset > 0) {
+    auto currentTime = dateTimeController.CurrentDateTime().time_since_epoch();
+
+    // Convert current time to a time_point and get hours and minutes since midnight
+    auto currentSeconds = std::chrono::duration_cast<std::chrono::seconds>(currentTime).count();
+    std::time_t t = currentSeconds;
+    std::tm localTime = *std::localtime(&t);
```

Comment:
I'd prefer if we used `std::chrono` instead of localtime

something like `midnight = std::chrono::floor<std::chrono::days>(currentTime)` to get the time object containing midnight of the specified day

https://stackoverflow.com/questions/61273498/number-of-seconds-since-midnight

beware: we can't use `std::chrono::system_clock::now()` or similar on PineTime. We have to use the DateTimeController to get the current time. The devious part is that `system_clock::now()` works in InfiniSim, but not in firmware

Reply:
nice! way simpler too, thanks.